ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,137 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Example of setInterval
+var init_interval
+
+// Record timer waiting to run
+var timer_queue = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const removeTimer = (key, DB) => {
+    try {
+        //Delete thread when thread exists
+        if (timer_queue[key] != undefined ) {
+            clearTimeout(timer_queue[key].timer)
+        }
+        delete timer_queue[key];
+        // An insurance mechanism to ensure that thread init_interval is active
+        init_secret_delete_thread(DB)
```

Comment:
this is a protection mechanism, if the_InitTimer in @line99 stopped unexpectedly, it will run again here

Reply:
this is a protection mechanism, if the_InitTimer in LINE99 stopped unexpectedly, it will run again here

Reply:
> this is a protection mechanism, if the_InitTimer in LINE99 stopped unexpectedly, it will run again here

If not stopped, what's impact if call again?

Reply:
no impact, because of line97

Reply:
That's fine.  Thanks.